godot-macros/src/derive_export.rs

```diff
@@ -0,0 +1,80 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use venial::{Declaration, StructFields};
+
+use crate::util::{bail, decl_get_info, DeclInfo};
+use crate::ParseResult;
+
+pub fn transform(decl: Declaration) -> ParseResult<TokenStream2> {
+    let DeclInfo { name, .. } = decl_get_info(&decl);
+
+    let enum_ = match decl {
+        Declaration::Enum(e) => e,
+        Declaration::Struct(s) => {
+            return bail!(s.tk_struct, "Export can only be derived on enums for now")
+        }
+        Declaration::Union(u) => {
+            return bail!(u.tk_union, "Export can only be derived on enums for now")
+        }
+        _ => unreachable!(),
+    };
+
+    let mut hint_string = String::new();
+    if enum_.variants.is_empty() {
+        return bail!(
+            enum_.name,
+            "In order to derive Export, enums must have at least one variant"
+        );
+    } else {
+        for (i, (enum_v, _)) in enum_.variants.inner.iter().enumerate() {
+            let v_name = enum_v.name.clone();
+            let v_disc = if let Some(c) = enum_v.value.clone() {
+                c.value
+            } else {
+                return bail!(
+                    v_name,
+                    "Property can only be derived on enums with explicit discriminants in all their variants"
+                );
+            };
+
+            hint_string += &format!(
+                "{v_name}:{}{}",
+                v_disc.to_string().trim_matches(['(', ')'].as_slice()),
+                if i < enum_.variants.inner.len() - 1 {
+                    ","
+                } else {
+                    ""
+                }
+            );
```

Comment:
Could you extract the format arguments into separate variables? Then you can substitute them inline like in `{vname}`.

Also, maybe [`join()`](https://doc.rust-lang.org/std/primitive.slice.html#method.join) helps with the intermediate comma?